core/dbt/task/runnable.py

```diff
@@ -390,8 +390,11 @@ def _mark_dependent_errors(self, node_id, result, cause):
         for dep_node_id in self.graph.get_dependent_nodes(node_id):
             self._skipped_children[dep_node_id] = cause
 
-    def populate_adapter_cache(self, adapter):
-        adapter.set_relations_cache(self.manifest)
+    def populate_adapter_cache(self, adapter, required_schemas: Set[BaseRelation] = None):
+        if flags.SELECTED_SCHEMA_CACHE is True:
```

Comment:
Feels like we can just remove this function and call `adapter.set_relations_cache` in places that call `populate_adapter_cache`

Reply:
`populate_adapter_cache`  here in `GraphRunnableTask` has been inherited by `RunTask` where it is mostly used. Moving this logic to `before_run` function would duplicate the same logic in `before_run` function in `GraphRunnableTask` and `RunTask` class.
If there is room to reorganize stuffs around, might make sense to do by someone more experienced with these code then me. This is the first time I am touching dbt-core :-)